source/adapters/opencl/memory.cpp

```diff
@@ -411,6 +412,11 @@ UR_APIEXPORT ur_result_t UR_APICALL urMemRetain(ur_mem_handle_t hMem) {
 }
 
 UR_APIEXPORT ur_result_t UR_APICALL urMemRelease(ur_mem_handle_t hMem) {
+  cl_context CLContext;
+  CL_RETURN_ON_FAILURE(clGetMemObjectInfo(cl_adapter::cast<cl_mem>(hMem), CL_MEM_CONTEXT,
+                         sizeof(CLContext), &CLContext, nullptr));
+  CL_RETURN_ON_FAILURE(clReleaseContext(CLContext));
```

Comment:
We should be consistent about using either the UR or CL entry points, at the moment above used `urContextRetain` but this uses `clReleaseContext`.

Reply:
I made it using cl api everywhere as that avoid some casts and would be more consistent across the entry-point implementation.